components/tooltip/__docs__/__snapshots__/storybook-stories.storyshot

```diff
@@ -1992,41 +1991,45 @@ exports[`DOM snapshots SLDSPopoverTooltip Button Group 1`] = `
         Edit
       </button>
       <span />
-    </div>
-    <div
-      className="slds-dropdown-trigger slds-dropdown-trigger_click slds-button_last"
-      id="options"
-      onClick={[Function]}
-      onFocus={null}
-      onKeyDown={[Function]}
-      onMouseEnter={null}
-      onMouseLeave={null}
+    </li>
+    <li
+      triggerClassName="slds-button_last"
```

Comment:
I'm getting this warning on the example added at http://localhost:9001/?selectedKind=SLDSPopoverTooltip&selectedStory=Button%20Group&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Factions%2Factions-panel

![Screen Shot 2019-05-14 at 1 56 03 AM](https://user-images.githubusercontent.com/1290832/57680703-7445b300-75eb-11e9-97fb-48c6341a915a.png)

The ButtonGroup is expecting `Button` child components which have a `triggerClassName` but, but `li` element tags do not. Please fix this warning.

Reply:
```if (zeroIndexLength > 0) {
		children = React.Children.map(props.children, (child, index) => {
			let newChild;
			if (index === zeroIndexLength) {
				newChild = React.cloneElement(child, {
					triggerClassName: 'slds-button_last',
				});
			}

			return newChild || child;
		});
	}
```
The error is being generated from the above LOC (present in `button-group`). Adding a type check to prevent the prop `triggerClassName` being added to `li` element will resolve the warning. Shall I go ahead with that ? @interactivellama

Reply:
Yes. I wouldn't recommend variable type checking though. Since `triggerClassName` is an implementation details of `SLDSButton`, you could check for that and only apply the prop if it's an instance of a `SLDSButton`.

Reply:
But we are adding the prop to the instance of `Tooltip` rather than `SLDSButton`. Because the children of the `ButtonGroup` are `Tooltip`s and not `Button`s

Reply:
Do you have any other suggestions in mind?

Reply:
You are exactly right. My mistake. The prop will only work on a `Tooltip` component. So we want to test if the child is a `Tooltip`. 

Here is a displayName test in another component: https://github.com/salesforce/design-system-react/blob/master/components/builder-header/nav.jsx#L33

Reply:
I have added the `displayName` test for the `Tooltip`

Reply:
It is creating issues elsewhere in the projects wherever `ButtonGroup` is being used. My proposed solution would be to replace `li` with `Tooltip` and let it be there.

Reply:
If you are referring to PageHeader, https://travis-ci.org/salesforce/design-system-react/builds/534113631#L7524 for instance.

Like code, Jest snapshot markup tests are not fixed in stone. Similar to version control such as "git", the snapshots are a way to track history over time (humans are very bad at this!) and know what has changed across time. If there is a bug or a better way to do something, updating the Jest DOM snapshots is a perfectly fine thing to do. 

In summary, it's OK to change the DOM snapshots, but only if it's a better solution. They are there so that you know what is changing and can decide if you really want to update the markup. Please review the changes across the library per the "failed" Jest tests visually and decide if updating the markup is best and report back.